common/src/main/java/org/opensearch/sdk/client/LocalClusterIndicesClient.java

```diff
@@ -189,7 +195,13 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(DeleteDat
     public CompletionStage<SearchDataObjectResponse> searchDataObjectAsync(SearchDataObjectRequest request, Executor executor) {
         return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<SearchDataObjectResponse>) () -> {
             log.info("Searching {}", Arrays.toString(request.indices()));
-            SearchResponse searchResponse = client.search(new SearchRequest(request.indices(), request.searchSourceBuilder())).actionGet();
+            SearchSourceBuilder searchSource = request.searchSourceBuilder();
+            QueryBuilder existingQuery = searchSource.query();
+            String tenantId = request.tenantId() != null ? request.tenantId() : CommonValue.DEFAULT_TENANT;
+            BoolQueryBuilder boolQuery = QueryBuilders.boolQuery().must(existingQuery == null ? new MatchAllQueryBuilder() : existingQuery);
+            boolQuery.filter(QueryBuilders.termQuery(CommonValue.TENANT_ID, tenantId));
+            searchSource.query(boolQuery);
+            SearchResponse searchResponse = client.search(new SearchRequest(request.indices(), searchSource)).actionGet();
```

Comment:
We should add timeout for `actionGet` since if the response pending too long, the threads in the pool can be exhausted.

Reply:
I appreciate the concern but I don't think a timeout is the correct answer.

Do you think these should all be rewritten with ActionListeners?

Reply:
What's the concern on adding timeout? Rewritten to with actionListeners would be better but I think it needs more effort.

Reply:
> What's the concern on adding timeout? 

What should the timeout be?  Where do we allow the user to configure it?

> Rewritten to with actionListeners would be better

It does use an action listener internally as part of how this call works.  The future completes when the action listener gets the callback.   Anything I do to wrap an action listener would essentially duplicate the existing code returning a future.

> but I think it needs more effort.

"it needs more effort" is an extremely vague code review comment and isn't really helpful letting me know where you think the problem is or what you think the solution should be.  Can you suggest a specific alternative here?

The entire code block is wrapped inside a `CompletableFuture.supplyAsync()`.  The "search" call returns the future immediately so the only delay in this entire code block is the `actionGet()` call and any additional effort to wrap it is redundant to the existing async methods.

Reply:
> What should the timeout be? Where do we allow the user to configure it?

This can be a cluster setting and allow user to dynamically configure it.

> "it needs more effort" is an extremely vague code review comment and isn't really helpful letting me know where you think the problem is or what you think the solution should be. Can you suggest a specific alternative here?

My comment was based on your `rewritten with ActionListeners` and my assumption from this `rewritten` is to change all the API in `SdkClientDelegate` which needs much more effort than adding timeout. E.g. change from
```
CompletionStage<PutDataObjectResponse> putDataObjectAsync(PutDataObjectRequest request, Executor executor);
``` 
to
```
void putDataObjectAsync(PutDataObjectRequest request, ActionListener listener);
```
If this assumption is not correct, please specify how the rewritten will be done, thanks.

Reply:
> This can be a cluster setting and allow user to dynamically configure it.

There's already a timeout setting available on the REST params that users can set per-query.

This client is just a wrapper around the `NodeClient` `search()` call.  The NodeClient doesn't have a timeout; users who have timeout concerns can wrap it around their call to this method.  In particular, the `CompletableFuture` returned from this method has a `get()` option with a timeout.  

> My comment was based on your rewritten with ActionListeners

I looked into that and realized that it added no value, so my comment was premature.

I had intentionally used the non-ActionListener version of the API, wrapped in an async execution where the caller had more control over which thread pool was used.

> If this assumption is not correct, please specify how the rewritten will be done, thanks.

At this stage I don't think I will rewrite with `ActionListener` at all.

The only thing I'm currently considering doing is wrapping the `actionGet()` call, and only that call, inside another `supplyAsync()`.

Reply:
LGTM if user can set timeout to this client.

Reply:
1. Search query can use `timeout` field in search query at shard level which will be passed through.
2. The surrounding `CompletableFuture` can use timeout fields to end that query early.